tests/components/drop_connect/test_sensor.py

```diff
@@ -34,286 +39,97 @@
+    data: str,
 ) -> None:
-    """Test DROP sensors for pump controllers."""
-    entry = config_entry_pump_controller()
-    entry.add_to_hass(hass)
-    assert await hass.config_entries.async_setup(entry.entry_id)
+    """Test DROP sensors."""
+    config_entry.add_to_hass(hass)
 
-    current_flow_sensor_name = "sensor.pump_controller_water_flow_rate"
-    assert hass.states.get(current_flow_sensor_name).state == STATE_UNKNOWN
-    psi_sensor_name = "sensor.pump_controller_current_water_pressure"
-    assert hass.states.get(psi_sensor_name).state == STATE_UNKNOWN
-    temp_sensor_name = "sensor.pump_controller_temperature"
-    assert hass.states.get(temp_sensor_name).state == STATE_UNKNOWN
+    with patch("homeassistant.components.drop_connect.PLATFORMS", [Platform.SENSOR]):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
 
-    async_fire_mqtt_message(
-        hass, TEST_DATA_PUMP_CONTROLLER_TOPIC, TEST_DATA_PUMP_CONTROLLER_RESET
-    )
-    await hass.async_block_till_done()
-    async_fire_mqtt_message(
-        hass, TEST_DATA_PUMP_CONTROLLER_TOPIC, TEST_DATA_PUMP_CONTROLLER
+    entity_entries = er.async_entries_for_config_entry(
+        entity_registry, config_entry.entry_id
     )
-    await hass.async_block_till_done()
-
-    current_flow_sensor = hass.states.get(current_flow_sensor_name)
-    assert current_flow_sensor
-    assert current_flow_sensor.state == "2.2"
-
-    psi_sensor = hass.states.get(psi_sensor_name)
-    assert psi_sensor
-    assert psi_sensor.state == "428.8538854"  # centibars
-
-    temp_sensor = hass.states.get(temp_sensor_name)
-    assert temp_sensor
-    assert temp_sensor.state == "20.4444444444444"  # °C
 
+    assert entity_entries
+    for entity_entry in entity_entries:
+        assert hass.states.get(entity_entry.entity_id).state == STATE_UNKNOWN
 
-async def test_sensors_ro_filter(
-    hass: HomeAssistant, mqtt_mock: MqttMockHAClient
-) -> None:
-    """Test DROP sensors for RO filters."""
-    entry = config_entry_ro_filter()
-    entry.add_to_hass(hass)
-    assert await hass.config_entries.async_setup(entry.entry_id)
-
-    tds_in_sensor_name = "sensor.ro_filter_inlet_tds"
-    assert hass.states.get(tds_in_sensor_name).state == STATE_UNKNOWN
-    tds_out_sensor_name = "sensor.ro_filter_outlet_tds"
-    assert hass.states.get(tds_out_sensor_name).state == STATE_UNKNOWN
-    cart1_sensor_name = "sensor.ro_filter_cartridge_1_life_remaining"
-    assert hass.states.get(cart1_sensor_name).state == STATE_UNKNOWN
-    cart2_sensor_name = "sensor.ro_filter_cartridge_2_life_remaining"
-    assert hass.states.get(cart2_sensor_name).state == STATE_UNKNOWN
-    cart3_sensor_name = "sensor.ro_filter_cartridge_3_life_remaining"
-    assert hass.states.get(cart3_sensor_name).state == STATE_UNKNOWN
-
-    async_fire_mqtt_message(hass, TEST_DATA_RO_FILTER_TOPIC, TEST_DATA_RO_FILTER_RESET)
+    async_fire_mqtt_message(hass, topic, reset)
```

Comment:
May be we should also assert the state after the reset message was received.

Reply:
The reset message was added to ensure that the sensors were in a state known to be different than the expected state after the 'normal' MQTT data payload was delivered.  Given that the mutable object inheritance issue has been fixed, the 'reset' MQTT payloads are likely unnecessary.

Reply:
Well that seems not always the case. For the other platforms I have added extra asserts after the reset payload.